scripts/generate_serialization.py

```diff
@@ -122,46 +141,79 @@ def replace_pointer(type):
     return re.sub('([a-zA-Z0-9]+)[*]', 'unique_ptr<\\1>', type)
 
 
-def get_serialize_element(property_name, property_id, property_key, property_type, is_optional, pointer_type):
-    write_method = 'WriteProperty'
+def get_serialize_element(
+    property_name, property_id, property_key, property_type, has_default, default_value, is_deleted, pointer_type
+):
     assignment = '.' if pointer_type == 'none' else '->'
-    default_argument = ''
-    if is_optional:
-        write_method = 'WritePropertyWithDefault'
-        default_argument = f', {property_type}()'  # TODO: allow this to be passed
+    default_argument = '' if default_value is None else f', {default_value}'
+    template = serialize_element
+    if has_default and is_deleted:
```

Comment:
Can we now also remove `optional` from all fields that are marked as such in the JSON? If so, could we do that in this PR, and remove `optional` from `supported_member_entries` in the generate script?